tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/CalendarEventReplyPerformer.scala

```diff
@@ -158,46 +157,119 @@ class CalendarEventMailReplyGenerator(val bodyPartContentGenerator: CalendarRepl
-          case _ => throw new IllegalArgumentException("The eml file must contain a text body.")
+class I18NCalendarEventReplyMessageGenerator(fileSystem: FileSystem, i18nEmlDirectory: String) extends CalendarReplyMessageGenerator {
+
+  import I18NCalendarEventReplyMessageGenerator._
+
+  override def getBasedMimeMessage(i18n: Locale, attendeeReply: AttendeeReply, requestCalendar: Calendar): SMono[MimeMessage] =
+    getBasedMimeMessage(attendeeReply, requestCalendar, evaluateMailTemplateFileName(attendeeReply.partStat, i18n))
+
+  def getBasedMimeMessage(attendeeReply: AttendeeReply, calendar: Calendar, emlFilename: String): SMono[MimeMessage] =
+    SMono.fromCallable(() => URI.create(i18nEmlDirectory).resolve(emlFilename).toString)
+      .doOnNext(validateTemplateFile)
+      .map(emlLocation => new MimeMessageWrapper(MimeMessageInputStreamSource.create(MailImpl.getId, fileSystem.getResource(emlLocation))))
+      .map(mimeMessage => decoratedMimeMessage(mimeMessage, getMustacheDataMap(attendeeReply, calendar)))
+      .subscribeOn(Schedulers.boundedElastic())
+
+  private def decoratedMimeMessage(originalMimeMessage: MimeMessage, mustacheDataMap: JavaMap[String, String]): MimeMessage = {
+    originalMimeMessage.getContent match {
+      case textBody: String =>
+        val decoratedBody = decorateTextBodyMessage(textBody, mustacheDataMap)
+        originalMimeMessage.setContent(decoratedBody, originalMimeMessage.getContentType)
+      case mimeMultipart: MimeMultipart =>
+        val partNumber = mimeMultipart.getCount
+        for (i <- 0 until partNumber) {
+          val bodyPart = mimeMultipart.getBodyPart(i)
+          if (bodyPart.getDisposition == null || bodyPart.getDisposition.equalsIgnoreCase(Part.INLINE)) {
+            if (bodyPart.isMimeType("text/plain") || bodyPart.isMimeType("text/html")) {
+              val decoratedBody = decorateTextBodyMessage(bodyPart.getContent.asInstanceOf[String], mustacheDataMap)
+              bodyPart.setContent(decoratedBody, bodyPart.getContentType)
+            }
+          }
         }
-        ReplyBodyPart(generateSubject(attendeeReply, calendar, Option(mimeMessage.getSubject)), bodyPart)
-      })
+      case _ => throw new IllegalArgumentException("The eml file must contain a text body.")
+    }
+    originalMimeMessage.setSubject(decorateSubjectMessage(originalMimeMessage.getSubject, mustacheDataMap))
+    originalMimeMessage.saveChanges()
+    originalMimeMessage
   }
 
-  private def generateSubject(attendeeReply: AttendeeReply, calendar: Calendar, statPrefix: Option[String] = None): String = {
+  private def validateTemplateFile(emlLocation: String): Unit =
+    try {
+      fileSystem.getFile(emlLocation)
+    } catch {
+      case e: FileNotFoundException => throw new IllegalArgumentException("Cannot find the template eml file: " + emlLocation, e)
+      case e: Exception => throw new IllegalArgumentException("Cannot read the template eml file: " + emlLocation, e)
```

Comment:
s"Cannot find the template eml file: $emlLocation"

s"Cannot read the template eml file: $emlLocation"